File: CinemaTicket.kt/src/main/kotlin/com/example/cinematicket/models/visitor.kt
package domain.models

enum class Gender {
    Male, Female, Other
}

enum class VisitorType {
    General,
    Member,
    SeniorMember,
    Senior,
    DisabledPerson,
    College,
    HighSchoolStudent,
    JuniorHighSchoolStudent,
    ElementaryStudent
}

public class Visitor(var id: String, var age: Int, var gender: Gender, var visitorType: VisitorType) {
}

File: CinemaTicket/src/domain/models/visitor.ts
import Gender from './gender';

export const VisitorType = {
  general: 'general',
  member: 'member',
  seniorMember: 'seniorMember',
  senior: 'senior',
  disabledPerson: 'disabledPerson',
  college: 'college',
  highSchoolStudent: 'highSchoolStudent',
  juniorHighSchoolStudent: 'juniorHighSchoolStudent',
  elementaryStudent: 'elementaryStudent',
};

export class Visitor {
    private id: string

    public age: number

    public gender: Gender

    public visitorType: string

    constructor(
      id: string,
      age: number,
      gender: Gender,
      visitorType: string,
    ) {
      this.id = id;
      this.age = age;
      this.gender = gender;
      this.visitorType = visitorType;
    }
}
